Buena noche, a continuación presento la solución a mi parcial final en su parte práctica.  
Soy Brian David Ruiz Arango con código de estudiante: 101202414489.  
Dataset correspondiente: Titanic-Dataset.csv

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv("Titanic-Dataset.csv")
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


En el paso anterior incio importando las librerias necesarias para trabajar, las defino con base en lo trabajado durante el semestre en la materia de Programación para Analítica de Datos.  
Luego leo el dataset del csv que me corresponde y lo aisgno a la variable para trabajar con él.

In [3]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


Miro cuales son los tipos de dato de cada una de las columnas y otra información adicional para ver que aspectos llaman la atención de estos datos e identificar que se puede hacer con ellos.  
Por ejemplo veo que la variable "Cabin" tiene demasiados nulos y creo que es una información no tan necesaria para el análisis que se me ocurre.  
La columna "Survived" está completa y tiene solo 2 opciones: 0 y 1. Con esto puedo concluir que es una variable binaria que me dice si alguien sobrevivió o no.  
Para mi análisis puedo identificar algunas columnas de interes, como "Sex", "Age", "Embarked".  
Algunas variables como "Name", "PassengerId" ó "Ticket" no me parecen muy infliyentes en el análisis que haré.  

In [6]:
df = df.drop(["PassengerId", "Name", "Ticket", "Cabin"], axis = 1)

Decido eliminar las columnas: "PassengerId", "Name", "ticket" y "Cabin".  
Porque como ya mencioné algunas de ellas no las veo útiles en el enfoque de mi análisis y porque "cabin" no tiene suficiente información para  
reemplazar o completar la cantidad de información faltante.  
Esto me parece importante para evitar ruido en mi dataset y poder trabajar más fácil y cómodo con él.

In [7]:
df["Age"] = df["Age"].fillna(df["Age"].mean())
df["Embarked"] = df["Embarked"].fillna(df["Embarked"].mode()[0])

En la información adicional de arriba se evidencia que hay columnas como "Age" y "Embarked" que contienen nulos y por esto es necesario completar la información ya que no son tantos los valores faltantes como para eliminar la columna o filas completas y así mantener la distribución de los datos lo mejor posible.  
El total de datos son 891 filas y los faltantes en cada una no superan el 20% de la inforamción.  
En edad la media es una buena manera de centralizar los datos y en embarque entendiendo que es el puerto donde se embarca reemplazar por el valor que más se repite.

In [8]:
df = pd.get_dummies(df, columns = ["Sex", "Embarked"], drop_first = True)

In [9]:
df.head()

,Survived,Pclass,Age,SibSp,Parch,Fare,Sex_male,Embarked_Q,Embarked_S
0,0,3,22.0,1,0,7.2500,True,False,True
1,1,1,38.0,1,0,71.2833,False,False,False
2,1,3,26.0,0,0,7.9250,False,False,True
3,1,1,35.0,1,0,53.1000,False,False,True
4,0,3,35.0,0,0,8.0500,True,False,True


In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Survived    891 non-null    int64  
 1   Pclass      891 non-null    int64  
 2   Age         891 non-null    float64
 3   SibSp       891 non-null    int64  
 4   Parch       891 non-null    int64  
 5   Fare        891 non-null    float64
 6   Sex_male    891 non-null    bool   
 7   Embarked_Q  891 non-null    bool   
 8   Embarked_S  891 non-null    bool   
dtypes: bool(3), float64(2), int64(4)
memory usage: 44.5 KB


Las variables categóricas, las que son de tipo texto, es mejor cambiarlas por numéricas o booleanas.  
Observamos que ahora todas las columnas tienen sus datos completos y que tenemos tipos de variables cómodos para trabajar en nuestro análisis, además cuando entrenemos el módelo o se quieran hacer cálculos, tener los tipos de variables en números o un tipo fácil de contar es más sencillo.